src/NumericalAlgorithms/SphericalHarmonics/Strahlkorper.cpp

```diff
@@ -34,27 +37,33 @@ Strahlkorper<Frame>::Strahlkorper(const size_t l_max, const size_t m_max,
 template <typename Frame>
 Strahlkorper<Frame>::Strahlkorper(
     const size_t l_max, const size_t m_max,
-    const DataVector& collocation_radii_or_spectral_coefs,
-    std::array<double, 3> center, const StrahlkorperConstructorData data_type)
+    const DataVector& radius_at_collocation_points,
+    std::array<double, 3> center)
     : l_max_(l_max),
       m_max_(m_max),
       ylm_(l_max, m_max),
       // clang-tidy: do not std::move trivially constructable types
-      center_(std::move(center)) {  //,  // NOLINT
-  if (data_type == StrahlkorperConstructorData::RadiusAtCollocationPoints) {
-    ASSERT(collocation_radii_or_spectral_coefs.size() == ylm_.physical_size(),
-           "Bad size " << collocation_radii_or_spectral_coefs.size()
-                       << ", expected " << ylm_.physical_size());
-    strahlkorper_coefs_ =
-        ylm_.phys_to_spec(collocation_radii_or_spectral_coefs);
-  } else if (data_type == StrahlkorperConstructorData::SpectralCoefficients) {
-    ASSERT(collocation_radii_or_spectral_coefs.size() == ylm_.spectral_size(),
-           "Bad size " << collocation_radii_or_spectral_coefs.size()
-                       << ", expected " << ylm_.spectral_size());
-    strahlkorper_coefs_ = collocation_radii_or_spectral_coefs;
-  } else {
-    ERROR("Unknown StrahlkorperConstructorData type.");
-  }
+      center_(std::move(center)),  // NOLINT
+      strahlkorper_coefs_(ylm_.phys_to_spec(radius_at_collocation_points)) {
+  ASSERT(radius_at_collocation_points.size() == ylm_.physical_size(),
+         "Bad size " << radius_at_collocation_points.size() << ", expected "
+                     << ylm_.physical_size());
+}
+
+template <typename Frame>
+Strahlkorper<Frame>::Strahlkorper(const size_t l_max, const size_t m_max,
+                                  const ModalVector& spectral_coefficients,
+                                  std::array<double, 3> center)
+    : l_max_(l_max),
+      m_max_(m_max),
+      ylm_(l_max, m_max),
+      // clang-tidy: do not std::move trivially constructable types
+      center_(std::move(center)) {  // NOLINT
+  ASSERT(spectral_coefficients.size() == strahlkorper_coefs_.size(),
```

Comment:
I was under the impression I didn't have to since I thought it would already be sized by [this line](https://github.com/sxs-collaboration/spectre/blob/7630eea057a54396e086f7a84cf7ba3fb948cca9/src/NumericalAlgorithms/SphericalHarmonics/Strahlkorper.hpp#L167) that sizes `strahlkorper_coefs_`, but let me know if I have that wrong

Reply:
Does that get evaluated after the l_max is set, or with the default value of 2? Seems safer to resize explicitly, but either way this should show up in tests.

Reply:
I looked this up and I think it's valid C++. The members are initialized in the order that they are declared, so it uses the correct l_max. Bit obscure though.

Reply:
I was going to say that we could change the assert to `ASSERT(spectral_coefficients.size() == ylm_.spectral_size()` but I guess the way it is now does also give the extra security that we know that `strahlkorper_coefs_` does not need to be resized, otherwise it would have failed the tests, as you said. But I agree, definitely an obscure detail

Reply:
Hmm I'm leaning towards removing the `ASSERT` and just explicitly doing `strahlkorper_coefs_.destructive_resize(...)` even though the current code is technically valid. This way it's very clear on the lines below that the container is the correct size and we don't have to rely on obscure knowledge on how default variables are initialized.

Reply:
So I was just about to add this but then I noticed that in the [docs](https://spectre-code.org/classVectorImpl.html#ad55c34f0c923e4d76ad94638c469c417) it says that if we expect it to be the right size, we shouldn't use it because it will be slow. I guess right now this constructor is only ever being called when we read in data, but nevertheless, it will be called during a time when we know that it will be correct every time and thus slower than it needs to be. Let me know if this changes your opinion or not!

Reply:
Oh wait, I completely misread it. It shouldn't be used when it's expected to resize - my bad! We're good 👍

Reply:
@knelli2 I amended the `ASSERT` so it's still useful in printing the error message it does during debug runs (just like the other constructor that takes the radius does) but also added the `destructive_resize` you requested